src/Std/Internal/Async/Tcp.lean

```diff
@@ -0,0 +1,120 @@
+/-
+Copyright (c) 2024 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sofia Rodrigues
+-/
+prelude
+import Std.Time
+import Std.Internal.UV
+import Std.Internal.Async.Basic
+import Std.Net.Addr
+
+namespace Std
+namespace Internal
+namespace IO
+namespace Async
+namespace Tcp
+
+open Std.Net
+
+/--
+A high-level wrapper around a TCP socket.
+-/
+structure Socket where
+  private ofNative ::
+    native : Internal.UV.Tcp.Socket
+
+namespace Socket
+
+/--
+Create a new TCP socket.
+-/
+@[inline]
+def mk : IO Socket := do
+  let native ← Internal.UV.Tcp.Socket.new
+  return Socket.ofNative native
+
+/--
+Bind the socket to the given address.
+-/
+@[inline]
+def bind (s : Socket) (addr : SocketAddress) : IO Unit :=
+  s.native.bind addr
+
+/--
+Connect the socket to the given address.
+-/
+@[inline]
+def connect (s : Socket) (addr : SocketAddress) : IO (AsyncTask Unit) :=
+  AsyncTask.ofPromise <$> s.native.connect addr
+
+/--
+Listen for incoming connections with the given backlog.
+-/
+@[inline]
+def listen (s : Socket) (backlog : UInt32) : IO Unit :=
+  s.native.listen backlog
+
+/--
+Accept an incoming connection.
+-/
+@[inline]
+def accept (s : Socket) : IO (AsyncTask Socket) := do
```

Comment:
Java has a split between `ServerSocket` and `Socket`, so that the accept function is defined on `ServerSocket` and returns a `Socket`. I wonder if this could also be an option for us.